cocos/core/utils/js-typed.ts

```diff
@@ -348,7 +348,21 @@ const REGEXP_STR = /%s/;
  * js.formatStr(a, b, c);
  * ```
  */
-export function formatStr (msg: string, ...subst: any[]): string {
+export function formatStr (msg: string, ...subst: StringSubstitution[]): string;
+/**
+ * @en
+ * A string tool to constructs a string from an arbitrary sequence of js object arguments.
+ * @zh
+ * 根据任意 js 对象参数序列构造一个字符串。
+ * @returns @en A new formatted string. @zh 格式化后的新字符串。
+ * @example
+ * ```
+ * import { js } from 'cc';
+ * js.formatStr({}, null, undefined);  // [object Object] null undefined
+ * ```
+ */
+export function formatStr (...data: unknown[]): string
+export function formatStr (msg: unknown, ...subst: unknown[]): string {
     if (arguments.length === 0) {
         return '';
     }
```

Comment:
we should support another formatStr signature, see this use case in unit test:
https://github.com/cocos/cocos-engine/blob/develop/tests/core/js.test.ts